recipes-extended/rsyslog/rsyslog_8.2206.0.bbappend

```diff
@@ -0,0 +1,16 @@
+PR = "r1"
+
+# We don't want silly logrotate deps
+# As no one is currently using it
+# The same is valid for systemd service
+# as we provide our-own
+RDEPENDS:${PN}:remove = "logrotate"
+RCONFLICTS:${PN} = ""
```

Comment:
This `RCONFLICTS` I believe is not needed if we configure the `VIRTUAL-RUNTIME_base-utils-syslog` system wide as described below.

`logrotate` is fine as quite "hardcoded' in original recipe :-/ `SYSTEMD_SERVICE` below also since the only alternative would be to totally disable systemd on system, which we don't want.

Reply:
Sorry, I'm not sure I got your suggestion properly.

- `logrotate` is (currently) not installed nor built at all, `rsyslog` demanded it, that's way I explicitly removed it
   I didn't want to get other packages in, if not absolutely necessary. Also considering that log dirs are read-only
   and actually we log to stdout (and let journald handle that)

- `SYSTEMD_SERVICE` Why should this require disabling systemd at all? Apparently the only effect (on rsyslog recipe) is to get rid of systemd-related files from the `rsyslog*.rpm` package, so basically we can an RPM containing only libs and binaries, without service configuration that we should.
If I recall properly (which Is not sure!) Without this I had an error when removing systemd files from the package (but for sure I may incurred in some different problem...)

- `RCONFLICTS` was removed because getting rid of systemd it actually doesn't conflict with anything.
Anyway the main reason was to avoid a sort of chicken-and-egg problem when updating existing HSCs which have `base-syslog` preinstalled.
In teory, rebuilding all the packages and letting `dnf` handle the thing *should* work but, while testing the single packages I kept having to throw `--allow-overwrite` to allow install of `rsyslog` and removal of `base-syslog`.
The sysupgrader can probably solve the issue by properly installing/updating/removing the packages but I'm not sure it's worth complicating that part

Reply:
We misunderstood each other :) For `logrotate` and `SYSTEMD_SERVICE` I was just confirming your solution was good (because logrotate didn't have any more elegant way to do, ie. the package doesn't support build features like some do to disable specific features, and because for not having the services the only alternative would be, but of course is not possible, remove systemd entirely)! So mine was just a confirmation comment :)

For rconflicts: indeed I've been testing mostly doing the image from scratch, I didn't try on a previous system. If that generates problems then indeed we should leave the RCONFLICTS.